### ISSUE TITLE
Add Louisiana deduction logic for 2021

### ISSUE THREAD

**Issue by martinholmer** [PolicyEngine/policyengine-us#3589]
The 0.615.0 version implements 2021 LA deductions incorrectly.  There is at least one problem.   The 2021 deduction of federal income taxes, for which every taxpayer was eligible, is not implemented.  The 2021 tax documents are referenced, but the Python code does not implement this feature of the LA income tax.

Also, note that the handling of LA itemized deductions changed significantly beginning in 2022.

**Comment by PavelMakarchuk** [comment]
@martinholmer thank you for the issue, 

I am not sure about the 2021 item. deductions difference, looking at the [2021 tax form](https://revenue.louisiana.gov/TaxForms/IT540i(2021)%20Instructions.pdf#page=3) this is what I believe is the process:
8A - itemized deductions amount 
8B - Federal standard deduction amount 
8C - standard deduction subtracted from the itemized deductions

![image](https://github.com/PolicyEngine/policyengine-us/assets/110687043/009aebce-796e-4e02-b05e-09d9e698cd5a)

(The itemization selection was implemented in the LA taxable income var)
![image](https://github.com/PolicyEngine/policyengine-us/assets/110687043/db878548-ccdd-48b9-a544-d90998f71573)

**Comment by martinholmer** [comment]
@PavelMakarchuk,  Thanks for the correction to my issue #3589, which I have recently edited to be correct.

But looking at your PR #3596, I think things are not yet completely correct.   When I look at the 2021 and 2022 LA forms/instructions, it seems to me that there was a big change (beginning in 2022) in how LA itemized deductions are calculated.   Among other changes it seems as if the direct itemized deduction of federal income taxes has been eliminated, and as a result, the numbering of lines 8 through line 10 have been changed in 2022.   Be sure you understand these changes and get logic and the documentation references correct for both 2021 and 2022.
